plugins/out_s3/s3.c

```diff
@@ -2198,9 +2268,18 @@ static void cb_s3_flush(struct flb_event_chunk *event_chunk,
         }
         else {
             /* Send upload directly without upload queue */
-            ret = send_upload_request(ctx, chunk, upload_file, m_upload_file,
-                                      event_chunk->tag,
-                                      flb_sds_len(event_chunk->tag));
+            if (!upload_file) {
+                ret = send_upload_request(ctx, chunk, upload_file,
+                                          m_upload_file,event_chunk->tag,
+                                          flb_sds_len(event_chunk->tag), &tms.tm.tv_sec);
+            }
+            else {
+                ret = send_upload_request(ctx, chunk, upload_file,
+                                          m_upload_file,event_chunk->tag,
+                                          flb_sds_len(event_chunk->tag),
+                                          upload_file->first_log_time);
+            }
+
```

Comment:
To simplify logic can you always pass in `chunk_first_log_time` to this function, and have the function decide whether to use the upload_file's time or the current time if upload_file is null or not?

Reply:
So you should always just do: 
```
send_upload_request(ctx, chunk, upload_file, m_upload_file,
                                      event_chunk->tag,
                                      flb_sds_len(event_chunk->tag),
                                      chunk_first_log_time
);
```

You need to add some logic to the send_upload request function and change the signature to 
```
send_upload_request(..., time_t chunk_first_log_time)
```